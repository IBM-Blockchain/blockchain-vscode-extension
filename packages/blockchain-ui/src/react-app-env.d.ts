/// <reference types="react-scripts" />

interface File {
    path: string;
}
