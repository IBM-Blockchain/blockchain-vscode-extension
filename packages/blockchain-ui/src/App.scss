@import '@carbon/colors/scss/colors';

#root{
    .app-container__tutorial-page {
        background-color: $gray-10;
        position: absolute;
        width: 100%;
        z-index: -1;
        box-shadow: inset 0px 164x #FFFFFF;
        top: 0;
        left: 0;

    }
}
