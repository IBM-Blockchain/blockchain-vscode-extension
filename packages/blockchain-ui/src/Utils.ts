import IVSCode from './interfaces/IVSCode';
declare const vscode: IVSCode;

class Utils {

    static changeRoute(newPath: string): void {
        dispatchEvent(new MessageEvent('message', {
            data: {
                path: newPath
            }
        }));
    }

    static postToVSCode(message: { command: string, data?: any }): void {

        if (!(window as any)['Cypress']) {
            vscode.postMessage(message);
        } else {
            window.postMessage(message, '*');
        }
    }

}

export default Utils;
