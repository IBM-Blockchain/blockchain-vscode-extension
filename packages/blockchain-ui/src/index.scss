@import '@carbon/colors/scss/colors';
@import '@carbon/themes/scss/themes';
@import '@carbon/type/scss/type';
@import '@carbon/type/scss/font-face/_mono';
@import '@carbon/type/scss/font-face/_sans';
@import './mixins/theme-mixins.scss';

@include carbon--type-reset();
@include carbon--font-face-mono();
@include carbon--font-face-sans();

html {

    .vscode-dark {
        @include carbon--theme($carbon--theme--g90) {

            #root {
                // Color will default to #000000 otherwise
                color: $text-01;        
                @include select-dark;
            }

            // text inputs on the create transaction screen
            #create-txn-form {
                @include text-inputs-dark;
            }

            // data tables on home page
            #recent-txns-table, #saved-txns-table {
                @include data-table-dark;
            }

            // InlineSelect components
            #smart-contract-select-wrapper {
                @include inline-select-dark;
            }

            // Checkbox
            @include checkbox-dark;

            //panel containing transaction output
            #output-panel {
                @include output-panel-dark;
            }
        }
    }

    .vscode-light {
        @include carbon--theme($carbon--theme--g10) {
            
            #root {
                color: $text-01;
            }
        }
    }
}
