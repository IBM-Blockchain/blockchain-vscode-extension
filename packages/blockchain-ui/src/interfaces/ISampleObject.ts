import ICategoryObject from './ICategoryObject';

interface ISampleObject {
    name: string;
    description: string;
    readme: string;
    category: ICategoryObject;
}

export default ISampleObject;
