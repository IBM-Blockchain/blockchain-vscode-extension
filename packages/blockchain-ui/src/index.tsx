import * as React from 'react';
import * as ReactDOM from 'react-dom';
import App from './App';
import './index.scss';

const app: any = ReactDOM.render(
    <App/>,
    document.getElementById('root')
);

// @ts-ignore
if (window.Cypress) {
    // @ts-ignore
    window.app = app;
}
