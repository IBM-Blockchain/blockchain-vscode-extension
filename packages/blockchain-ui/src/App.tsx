import * as React from 'react';
import './App.scss';
import { HashRouter as Router, Route, Redirect } from 'react-router-dom';
import TransactionHome from './components/TransactionHome/TransactionHome';
import TransactionCreate from './components/TransactionCreate/TransactionCreate';
import ISmartContract from './interfaces/ISmartContract';
import Utils from './Utils';

interface AppState {
    redirectPath: string;
    gatewayName: string;
    activeSmartContract: ISmartContract;
    smartContracts: Array<ISmartContract>;
    transactionOutput: string;
}

class App extends React.Component<{}, AppState> {
    constructor(props: {}) {
        super(props);
        this.state = {
            redirectPath: '',
            gatewayName: '',
            activeSmartContract: {
                name: '',
                version: '',
                channel: '',
                label: '',
                transactions: [],
                namespace: ''
            },
            smartContracts : [],
            transactionOutput: ''
        };
        this.switchSmartContract = this.switchSmartContract.bind(this);
        this.postMessageHandler = this.postMessageHandler.bind(this);
    }

    componentDidMount(): void {
        window.addEventListener('message', (event: MessageEvent) => {
            if (event.data.output) {
                this.setState({
                    transactionOutput: event.data.output
                });
            } else {
                this.setState({
                    redirectPath: event.data.path,
                    gatewayName: event.data.state ? event.data.state.gatewayName : this.state.gatewayName,
                    activeSmartContract: event.data.state ? event.data.state.activeSmartContract : this.state.activeSmartContract,
                    smartContracts: event.data.state ? event.data.state.smartContracts : this.state.smartContracts,
                });
            }
        });
    }

    switchSmartContract(newActiveContractLabel: string): void {
        const smartContracts: Array<ISmartContract> = this.state.smartContracts;
        this.setState({
            activeSmartContract: smartContracts.find((obj: ISmartContract) => obj.label === newActiveContractLabel) as ISmartContract,
            smartContracts: smartContracts
        });
    }

    postMessageHandler(command: string, data?: any): void {
        if (data === undefined) {
            data = {
                gatewayName: this.state.gatewayName,
                activeSmartContract: this.state.activeSmartContract,
                smartContracts: this.state.smartContracts
            };
        }

        Utils.postToVSCode({
            command: command,
            data: data
        });
    }

    render(): JSX.Element {
        if (this.state.redirectPath === '') {
            return <div></div>;
        } else {
            return (
                <Router>
                    <div>
                        <Route render={(): JSX.Element => <Redirect push to={this.state.redirectPath}/>}></Route>
                        <Route exact path='/transaction' render={(): JSX.Element =>
                            <TransactionHome gatewayName={this.state.gatewayName} activeSmartContract={this.state.activeSmartContract} smartContracts={this.state.smartContracts}
                                switchSmartContract={this.switchSmartContract} postMessageHandler={this.postMessageHandler}/>}>
                        </Route>
                        <Route exact path='/transaction/create' render={(): JSX.Element =>
                            <TransactionCreate activeSmartContract={this.state.activeSmartContract} transactionOutput={this.state.transactionOutput}
                                postMessageHandler={this.postMessageHandler}/>}>
                        </Route>
                    </div>
                </Router>
            );
        }
    }
}

export default App;
