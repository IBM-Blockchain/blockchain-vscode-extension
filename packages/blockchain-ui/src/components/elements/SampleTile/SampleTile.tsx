import React, { Component } from 'react';
import { CheckmarkOutline16, MisuseOutline16 } from '@carbon/icons-react';
import { Button } from 'carbon-components-react';
import ISampleObject from '../../../interfaces/ISampleObject';

import { ExtensionCommands } from '../../../ExtensionCommands';
import Utils from '../../../Utils';
import '../../../styles/gallery.scss';

interface IProps {
    repositoryName: string;
    sampleObject: ISampleObject;
}

class SampleTile extends Component <IProps> {

    constructor(props: Readonly<IProps>) {
        super(props);

        this.openSampleHandler = this.openSampleHandler.bind(this);
    }

    openSampleHandler(): void {
        Utils.postToVSCode({
            command: ExtensionCommands.OPEN_SAMPLE_PAGE,
            data: [
                this.props.repositoryName,
                this.props.sampleObject.name,
            ]
        });
    }

    render(): JSX.Element {
        return (
            <div className='tab-container'>
                <div className='title'>
                    {this.props.sampleObject.name}
                </div>
                <div className='badges'>

                    <div className='badge-container'>
                        {this.props.sampleObject.category.contracts && this.props.sampleObject.category.contracts.length > 0 ?
                            <CheckmarkOutline16 className='icon'></CheckmarkOutline16> :
                            <MisuseOutline16 className='icon'></MisuseOutline16>
                        }
                        <span className='text'>Sample contracts</span>
                    </div>

                    <div className='badge-container'>
                        {this.props.sampleObject.category.applications && this.props.sampleObject.category.applications.length > 0 ?
                            <CheckmarkOutline16 className='icon'></CheckmarkOutline16> :
                            <MisuseOutline16 className='icon'></MisuseOutline16>
                        }
                        <span className='text'>Sample applications</span>
                    </div>

                </div>
                <div className='objectives'>
                    {this.props.sampleObject.description}
                </div>
                <div className='button-container'>
                    <Button className='button' kind='primary' size='default' onClick={this.openSampleHandler}>Open sample</Button>
                </div>
            </div>
        );
    }
}

export default SampleTile;
