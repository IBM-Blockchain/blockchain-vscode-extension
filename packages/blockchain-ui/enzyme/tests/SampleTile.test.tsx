import React from 'react';
import renderer from 'react-test-renderer';
import { mount, ReactWrapper } from 'enzyme';
import chai from 'chai';
import sinon from 'sinon';
import sinonChai from 'sinon-chai';

import Utils from '../../src/Utils';
import { ExtensionCommands } from '../../src/ExtensionCommands';
import ISampleObject from '../../src/interfaces/ISampleObject';
import SampleTile from '../../src/components/elements/SampleTile/SampleTile';
import { CheckmarkOutline16, MisuseOutline16 } from '@carbon/icons-react';

chai.should();
chai.use(sinonChai);

// tslint:disable no-unused-expression
interface IProps {
    sampleObject: ISampleObject;
    repositoryName: string;
}

describe('SampleTile component', () => {
    let mySandBox: sinon.SinonSandbox;
    let postToVSCodeStub: sinon.SinonStub;

    beforeEach(() => {
        mySandBox = sinon.createSandbox();
        postToVSCodeStub = mySandBox.stub(Utils, 'postToVSCode').resolves();
    });

    afterEach(() => {
        mySandBox.restore();
    });

    const sampleObject: ISampleObject = {
        name: 'FabCar',
        description: 'Basic sample based on cars: the "hello world" of Hyperledger Fabric samples.',
        readme: 'https://raw.githubusercontent.com/hyperledger/fabric-samples/master/README.md',
        category: {
            contracts: [
                {
                    name: 'FabCar Contract',
                    languages: [
                        {
                            type: 'Go',
                            version: '1.0.0',
                            workspaceLabel: 'fabcar-contract-go',
                            remote: {
                                branch: 'master',
                                path: 'chaincode/fabcar/go'
                            }
                        },
                        {
                            type: 'Java',
                            version: '1.0.0',
                            workspaceLabel: 'fabcar-contract-java',
                            remote: {
                                branch: 'master',
                                path: 'chaincode/fabcar/java'
                            }
                        },
                        {
                            type: 'JavaScript',
                            version: '1.0.0',
                            workspaceLabel: 'fabcar-contract-javascript',
                            remote: {
                                branch: 'master',
                                path: 'chaincode/fabcar/javascript'
                            },
                            onOpen: [
                                {
                                    message: 'Installing Node.js dependencies ...',
                                    command: 'npm',
                                    arguments: ['install']
                                }
                            ]
                        },
                        {
                            type: 'TypeScript',
                            version: '1.0.0',
                            workspaceLabel: 'fabcar-contract-typescript',
                            remote: {
                                branch: 'master',
                                path: 'chaincode/fabcar/typescript'
                            },
                            onOpen: [
                                {
                                    message: 'Installing Node.js dependencies ...',
                                    command: 'npm',
                                    arguments: ['install']
                                }
                            ]
                        }
                    ]
                }
            ],
            applications: [
                {
                    name: 'Java Application',
                    type: 'Web',
                    version: '1.0.0',
                    language: 'Java',
                    readme: 'https://github.com/hyperledger/fabric-samples',
                    workspaceLabel: 'fabcar-app-java',
                    remote: {
                        branch: 'master',
                        path: 'fabcar/java'
                    }
                },
                {
                    name: 'JavaScript Application',
                    type: 'Web',
                    version: '1.0.0',
                    language: 'JavaScript',
                    readme: 'https://github.com/hyperledger/fabric-samples',
                    workspaceLabel: 'fabcar-app-javascript',
                    remote: {
                        branch: 'master',
                        path: 'fabcar/javascript'
                    },
                    onOpen: [
                        {
                            message: 'Installing Node.js dependencies ...',
                            command: 'npm',
                            arguments: ['install']
                        }
                    ]
                },
                {
                    name: 'TypeScript Application',
                    type: 'Web',
                    version: '1.0.0',
                    language: 'TypeScript',
                    readme: 'https://github.com/hyperledger/fabric-samples',
                    workspaceLabel: 'fabcar-app-typescript',
                    remote: {
                        branch: 'master',
                        path: 'fabcar/typescript'
                    },
                    onOpen: [
                        {
                            message: 'Installing Node.js dependencies ...',
                            command: 'npm',
                            arguments: ['install']
                        }
                    ]
                }
            ]
        }
    };

    it('should render the expected snapshot', () => {
        const component: any = renderer
            .create(<SampleTile sampleObject={sampleObject} repositoryName='fabric-samples'/>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });

    it(`should post a message to VS Code when the 'Open sample' button is clicked`, () => {
        const component: ReactWrapper<IProps> = mount(<SampleTile sampleObject={sampleObject} repositoryName='fabric-samples'/>);
        component.find('button').at(0).simulate('click');
        postToVSCodeStub.should.have.been.calledOnceWithExactly({
            command: ExtensionCommands.OPEN_SAMPLE_PAGE,
            data: [
                'fabric-samples',
                sampleObject.name
            ]
        });
        component.find('button').at(0).hasClass('bx--btn--primary').should.equal(true);
    });

    it(`should display to user that no smart contract exist`, () => {
        const noContractObject: ISampleObject = {
            name: 'FabCar',
            description: 'Basic sample based on cars: the "hello world" of Hyperledger Fabric samples.',
            readme: 'https://raw.githubusercontent.com/hyperledger/fabric-samples/master/README.md',
            category: {
                contracts: [],
                applications: [
                    {
                        name: 'Java Application',
                        type: 'Web',
                        version: '1.0.0',
                        language: 'Java',
                        readme: 'https://github.com/hyperledger/fabric-samples',
                        workspaceLabel: 'fabcar-app-java',
                        remote: {
                            branch: 'master',
                            path: 'fabcar/java'
                        }
                    },
                    {
                        name: 'JavaScript Application',
                        type: 'Web',
                        version: '1.0.0',
                        language: 'JavaScript',
                        readme: 'https://github.com/hyperledger/fabric-samples',
                        workspaceLabel: 'fabcar-app-javascript',
                        remote: {
                            branch: 'master',
                            path: 'fabcar/javascript'
                        },
                        onOpen: [
                            {
                                message: 'Installing Node.js dependencies ...',
                                command: 'npm',
                                arguments: ['install']
                            }
                        ]
                    },
                    {
                        name: 'TypeScript Application',
                        type: 'Web',
                        version: '1.0.0',
                        language: 'TypeScript',
                        readme: 'https://github.com/hyperledger/fabric-samples',
                        workspaceLabel: 'fabcar-app-typescript',
                        remote: {
                            branch: 'master',
                            path: 'fabcar/typescript'
                        },
                        onOpen: [
                            {
                                message: 'Installing Node.js dependencies ...',
                                command: 'npm',
                                arguments: ['install']
                            }
                        ]
                    }
                ]
            }
        };

        const component: ReactWrapper<IProps> = mount(<SampleTile sampleObject={noContractObject} repositoryName='fabric-samples'/>);
        component.find('.badge-container').at(0).find(CheckmarkOutline16).length.should.equal(0);
        component.find('.badge-container').at(0).find(MisuseOutline16).length.should.equal(1);

        component.find('.badge-container').at(1).find(CheckmarkOutline16).length.should.equal(1);
        component.find('.badge-container').at(1).find(MisuseOutline16).length.should.equal(0);
    });

    it(`should display to user that no applications exist`, () => {
        const noApplicationObject: ISampleObject = {
            name: 'FabCar',
            description: 'Basic sample based on cars: the "hello world" of Hyperledger Fabric samples.',
            readme: 'https://raw.githubusercontent.com/hyperledger/fabric-samples/master/README.md',
            category: {
                contracts: [
                    {
                        name: 'FabCar Contract',
                        languages: [
                            {
                                type: 'Go',
                                version: '1.0.0',
                                workspaceLabel: 'fabcar-contract-go',
                                remote: {
                                    branch: 'master',
                                    path: 'chaincode/fabcar/go'
                                }
                            },
                            {
                                type: 'Java',
                                version: '1.0.0',
                                workspaceLabel: 'fabcar-contract-java',
                                remote: {
                                    branch: 'master',
                                    path: 'chaincode/fabcar/java'
                                }
                            },
                            {
                                type: 'JavaScript',
                                version: '1.0.0',
                                workspaceLabel: 'fabcar-contract-javascript',
                                remote: {
                                    branch: 'master',
                                    path: 'chaincode/fabcar/javascript'
                                },
                                onOpen: [
                                    {
                                        message: 'Installing Node.js dependencies ...',
                                        command: 'npm',
                                        arguments: ['install']
                                    }
                                ]
                            },
                            {
                                type: 'TypeScript',
                                version: '1.0.0',
                                workspaceLabel: 'fabcar-contract-typescript',
                                remote: {
                                    branch: 'master',
                                    path: 'chaincode/fabcar/typescript'
                                },
                                onOpen: [
                                    {
                                        message: 'Installing Node.js dependencies ...',
                                        command: 'npm',
                                        arguments: ['install']
                                    }
                                ]
                            }
                        ]
                    }
                ],
                applications: []
            }
        };

        const component: ReactWrapper<IProps> = mount(<SampleTile sampleObject={noApplicationObject} repositoryName='fabric-samples'/>);
        component.find('.badge-container').at(0).find(CheckmarkOutline16).length.should.equal(1);
        component.find('.badge-container').at(0).find(MisuseOutline16).length.should.equal(0);

        component.find('.badge-container').at(1).find(CheckmarkOutline16).length.should.equal(0);
        component.find('.badge-container').at(1).find(MisuseOutline16).length.should.equal(1);
    });

});
