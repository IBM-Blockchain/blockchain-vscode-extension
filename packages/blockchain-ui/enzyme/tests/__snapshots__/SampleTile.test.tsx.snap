// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SampleTile component should render the expected snapshot 1`] = `
<div
  className="tab-container"
>
  <div
    className="title"
  >
    FabCar
  </div>
  <div
    className="badges"
  >
    <div
      className="badge-container"
    >
      <svg
        aria-hidden={true}
        className="icon"
        focusable="false"
        height={16}
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 32 32"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16,2A14,14,0,1,0,30,16,14,14,0,0,0,16,2Zm0,26A12,12,0,1,1,28,16,12,12,0,0,1,16,28Z"
        />
        <polygon
          points="14 21.5 9 16.54 10.59 14.97 14 18.35 21.41 11 23 12.58 14 21.5"
        />
      </svg>
      <span
        className="text"
      >
        Sample contracts
      </span>
    </div>
    <div
      className="badge-container"
    >
      <svg
        aria-hidden={true}
        className="icon"
        focusable="false"
        height={16}
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 32 32"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16,2A14,14,0,1,0,30,16,14,14,0,0,0,16,2Zm0,26A12,12,0,1,1,28,16,12,12,0,0,1,16,28Z"
        />
        <polygon
          points="14 21.5 9 16.54 10.59 14.97 14 18.35 21.41 11 23 12.58 14 21.5"
        />
      </svg>
      <span
        className="text"
      >
        Sample applications
      </span>
    </div>
  </div>
  <div
    className="objectives"
  >
    Basic sample based on cars: the "hello world" of Hyperledger Fabric samples.
  </div>
  <div
    className="button-container"
  >
    <button
      className="button bx--btn bx--btn--primary"
      disabled={false}
      onClick={[Function]}
      tabIndex={0}
      type="button"
    >
      Open sample
    </button>
  </div>
</div>
`;
